source/velocity_boundary_conditions/gplates.cc

```diff
@@ -673,13 +673,13 @@ namespace aspect
                 first_data_file_number
                 + static_cast<unsigned int> (time_steps_since_start);
 
-              const bool load_both_files = std::abs(current_file_number - old_file_number) >= 1;
+              const bool load_both_files = std::abs(static_cast<int>(current_file_number - old_file_number)) >= 1;
```

Comment:
I am not sure this way everything still works as intended. There are just too many special cases (files starting from x > 0 or from 0, indices increasing with time or decreasing). I think this problem arises from my initial assumption that file numbers may not be negative, but thinking about it now, it seems that in fact they could. Somebody could name their files file_001, file_000, file_-001 to indicate this data is prescribing "the future" or "something before my plate motion model starts". So maybe the simplest solution is to do not make the `unsigned` assumption at all and handle everything as `int`. I can prepare a patch for this pull request later today (I think I know all the places this should affect).